WireUI/Tests/WireMainNavigationUITests/Containers/MainTabBarControllerTests.swift

```diff
@@ -23,67 +23,72 @@ import XCTest
 
 final class MainTabBarControllerTests: XCTestCase {
 
+    private var sut: MainTabBarController<
+        MockConversationListViewController,
+        MockConversationViewController<PreviewConversationModel>
+    >!
     private var snapshotHelper: SnapshotHelper!
 
-    override func setUp() {
+    @MainActor
+    override func setUp() async throws {
+        sut = .init()
         snapshotHelper = .init()
             .withSnapshotDirectory(SnapshotTestReferenceImageDirectory)
     }
 
     override func tearDown() {
         snapshotHelper = nil
+        sut = nil
     }
 
     @MainActor
-    func testConversationListAndConversationIsInstalled() throws {
+    func testConversationListIsInstalled() throws {
         // Given
         let conversationList = MockConversationListViewController()
-        let conversation = UIViewController()
-        let sut = MainTabBarController<MockConversationListViewController>()
 
         // When
-        sut.conversations = (conversationList, conversation)
+        sut.conversationList = conversationList
 
         // Then
-        let navigationController = try XCTUnwrap(sut.viewControllers?[1] as? UINavigationController)
-        XCTAssertEqual(navigationController.viewControllers, [conversationList, conversation])
+        let navigationController = try XCTUnwrap(sut.viewControllers?[1] as? UINavigationController) // TODO: [WPB-6647] fix index for navigation overhaul
+        XCTAssertEqual(navigationController.viewControllers, [conversationList])
     }
 
     @MainActor
-    func testConversationIsReleased() throws {
+    func testConversationIsReleased() async throws {
         // Given
-        let sut = MainTabBarController<MockConversationListViewController>()
-        sut.conversations = (.init(), .init())
-        let navigationController = try XCTUnwrap(sut.viewControllers?[1] as? UINavigationController)
+        weak var weakConversationList: MockConversationListViewController?
+        sut.conversationList = {
+            let conversationList = MockConversationListViewController()
+            weakConversationList = conversationList
+            return conversationList
+        }()
 
         // When
-        navigationController.popViewController(animated: false)
+        sut.conversationList = nil
 
         // Then
-        let (conversationList, conversation) = try XCTUnwrap(sut.conversations)
-        XCTAssertNotNil(conversationList)
-        XCTAssertNil(conversation)
+        await Task.yield()
```

Comment:
I am not sure to understand what we're doing here

Reply:
I wanted to ensure that assigning `nil` to any of the properties actually releases the previous reference (since internally the reference needs to be removed from the view controller hierarchy).